components/infobox/wikis/tetris/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,61 @@
+---
+-- @Liquipedia
+-- wiki=tetris
+-- page=Module:Infobox/Person/Player/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local Role = require('Module:Role')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local Player = Lua.import('Module:Infobox/Person', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local _role
+local _role2
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'role' then
+		_role = Role.run({role = _args.role})
+		_role2 = Role.run({role = _args.role2})
+		return {
+			Cell{name = _role2.display and 'Roles' or 'Role', content = {_role.display, _role2.display}}
+		}
+	end
+
+	return widgets
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:adjustLPDB(lpdbData)
+	lpdbData.extradata.isplayer = _role.isPlayer or 'true'
+	lpdbData.extradata.role = _role.role
+	lpdbData.extradata.role2 = _role2.role
```

Comment:
Are these extradata required? What's the use? Do we need to move these to commons? Can't `player.type` be used instead of `isPlayer`?

Reply:
From what I check role maybe needed to make this one shows the non-player roles <https://liquipedia.net/tetris/Portal:Players/Americas> - <https://liquipedia.net/commons/index.php?title=Module:Portal_players&action=edit>

not sure about isplayer but I can assume that maybe removable (try removing this role = person with Caster role still show under Staff section in the portal link, which is what supposed to be)

![image](https://user-images.githubusercontent.com/88981446/217283174-10f31616-08e6-4c42-880d-34aabc1f4209.png)